MERLin/MERLinTests/Tests/Events/ViewControllerEventsTests.swift

```diff
@@ -11,194 +11,74 @@ import RxSwift
 import RxTest
 import XCTest
 
-class MockPageViewController: UIViewController, PageRepresenting {
-    var pageName: String = "MockPage"
-    var section: String = "Test"
-    var pageType: String = "Mock"
-}
-
 class ViewControllerEventsTests: XCTestCase {
     var disposeBag: DisposeBag!
-    var module: MockModule!
+    var viewController: MockViewController!
     var scheduler: TestScheduler!
     
     override func setUp() {
         super.setUp()
         disposeBag = DisposeBag()
-        let context = ModuleContext(routingContext: "mock", building: MockModule.self)
-        module = MockModule(usingContext: context)
+        viewController = MockViewController()
         scheduler = TestScheduler(initialClock: 0)
     }
     
     override func tearDown() {
         disposeBag = nil
-        module = nil
+        viewController = nil
         scheduler = nil
         super.tearDown()
     }
     
     func makeObserverAndSubscribe() -> TestableObserver<ViewControllerEvent> {
         let observer = scheduler.createObserver(ViewControllerEvent.self)
         
-        module.viewControllerEvent
+        viewController.events
             .subscribe(observer)
             .disposed(by: disposeBag)
         
         return observer
     }
     
     func testThatItCanStoreTheObservable() {
-        let observable = module.viewControllerEvent
-        XCTAssertTrue(observable === module.viewControllerEvent)
-    }
-    
-    func testThatInitialStateIsUninitialised() {
-        let observer = makeObserverAndSubscribe()
-        scheduler.start()
-        
-        let expected = [
-            Recorded.next(0, ViewControllerEvent.uninitialized)
-        ]
-        
-        XCTAssertEqual(observer.events, expected)
-    }
-    
-    func testThatStateChangeToInitialised() {
-        let observer = makeObserverAndSubscribe()
-        scheduler.scheduleAt(1) {
-            _ = self.module.prepareRootViewController()
-        }
-        
-        scheduler.start()
-        
-        let expected = [
-            Recorded.next(0, ViewControllerEvent.uninitialized),
-            .next(1, .initialized),
-            .completed(1) // ViewControlelr dies so it causes the completed.
-        ]
-        
-        XCTAssertEqual(observer.events, expected)
+        let observable = viewController.events
+        XCTAssertTrue(observable === viewController.events)
```

Comment:
I think this test can be removed.

Reply:
Being an objective c runtime extension I want to make sure that the observable is not recreated at each invocation.

It's to make sure that the get associated object is used properly

Reply:
👍